docs/config-file.md

```diff
@@ -12,30 +12,84 @@ export default async function () {
 }
 ```
 
-This function must return an object with the following shape:
+This function must return an object with protocol specific configuration which would look like this: 
 
+
+```js
+export default async function () {
+  return {
+    websocket: {},
+    mqtt: {},
+    cluster: {}
+  }
+}
+
+```
+
+Here is an example of the `glee.config.js` file for refence: 
 ```js
 export default async function () {
   return {
     websocket: {
-      httpServer: customServer, // A custom HTTP server of your own.
-      adapter: 'native', // Default. Can also be 'socket.io' or a reference to a custom adapter.
-      port: process.env.PORT,
+      server: {
+        httpServer: customServer, // A custom HTTP server of your own.
+        adapter: "native", // Default. Can also be 'socket.io' or a reference to a custom adapter.
+        port: process.env.PORT,
+      },
+      client: {
+        authentication: {
+          http: () => {
+            const token = process.env.TOKEN
+            return token
+          }
+        }
+      }
     },
     cluster: {
-      adapter: 'redis',
-      name: 'cluster', // Default. Name of your cluster.
-      url: 'redis://localhost:6379', // Server URL used by adapter for clustering
+      adapter: "redis",
+      name: "cluster", // Default. Name of your cluster.
+      url: "redis://localhost:6379", // Server URL used by adapter for clustering
+    },
+    mqtt: {
+      authentication: {
+        cert: () => fs.readFileSync('./cert')
+        clientId: '123',
+        userPassword: {
+          username: 'user1',
+          password: 'pass12'
+        }
+      }
     }
-  }
+  };
 }
 ```
+Every protocol has different configuration needs so each protocol has unique configurations:
+
+### Websockets
+|Field|Description|
+|--|--|
+|websocket.server|Websocket server specific configuration|
+|websocet.client| Websocket client specific configuration|
+|wesocket.server.adapter| The Glee adapter to use for the WebSocket server. Defaults to a "native" WebSocket implementation. Other allowed values are `socket.io` (to use the [Socket.IO](https://socket.io/) Glee adapter) or a reference to a custom adapter.|
+|websocket.server.httpServer|  A custom HTTP server of your own. E.g., an [Express](https://expressjs.com/en/4x/api.html) server or any object that implements the [http.Server](https://nodejs.org/api/http.html#http_class_http_server) interface.   |
+|webocket.server.port| The port to use when binding the WebSocket server. This is useful when your server is behind a proxy and the port exposed for consumption is not the same as the port your application should be bound to. Defaults to the port specified in the selected AsyncAPI server.|
+|websocket.client.authentication.http| The http token to send for authentication of glee application.
```

Comment:
Shouldn't this allow for other types of authentication too? I sometimes use a bearer token, and other times I use an API token/key. Depending on the server, it may require me to pass a token like `Authentication: bearer [TOKEN]`, `Authentication: token [TOKEN]`, `Authentication: api-key [token]`, etc. I think being able to say how we authenticate (using a header or a query param) and customize the header value is the minimum required here.

Reply:
I am keeping on one kind of authentication for this PR, since this PR just focuses on updating the config file, In #319 I will add support for all the authentication methods supported for websocket.

Reply:
Have removed the `websocket.client` documentation for now.